pontoon/base/views.py

```diff
@@ -1224,20 +1228,19 @@ def save_to_transifex(request, template=None):
 
 
 @login_required(redirect_field_name='', login_url='/403')
+@require_POST
 @transaction.atomic
-def quality_checks_switch(request):
-    """Turn quality checks on/off for the current user."""
-    log.debug("Turn quality checks on/off for the current user.")
+def toggle_user_profile_attribute(request, attribute):
+    profile = request.user.profile
 
-    if request.method != 'POST':
-        log.error("Non-POST request")
-        raise Http404
+    if not hasattr(profile, attribute):
```

Comment:
I like the idea of generalization of such things, but in this case i have this fear that at some point this view could be easily abused if user could "toggle" any field in model. I mean, if there was a property similar to "is_superuser".